src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1256,14 +1257,21 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
         if (method_exists(PropertyAccessor::class, 'createCache')) {
             $propertyAccessDefinition = $container->register('cache.property_access', AdapterInterface::class);
             $propertyAccessDefinition->setPublic(false);
-            $propertyAccessDefinition->setFactory(array(PropertyAccessor::class, 'createCache'));
-            $propertyAccessDefinition->setArguments(array(null, null, $version, new Reference('logger', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)));
             $propertyAccessDefinition->addTag('cache.pool', array('clearer' => 'cache.default_clearer'));
             $propertyAccessDefinition->addTag('monolog.logger', array('channel' => 'cache'));
+
+            if (!$container->getParameter('kernel.debug')) {
+                $propertyAccessDefinition->setFactory(array(PropertyAccessor::class, 'createCache'));
+                $propertyAccessDefinition->setArguments(array(null, null, $version, new Reference('logger', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)));
+            } else {
+                $propertyAccessDefinition->setClass(ArrayAdapter::class);
+                $propertyAccessDefinition->setArguments(array(0));
```

Comment:
Removing this line gives:

> Symfony\Component\DependencyInjection\Exception\OutOfBoundsException: Cannot replace arguments if none have been configured yet.

>symfony/symfony/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php:74

`CachePoolPass.php:74` tries to replace arguments that are not set yet, for ArrayAdapter at least. Should it be fixed?